packages/nvidia_nat_core/tests/nat/profiler/prediction_trie/test_trie_builder.py

```
@@ -272,3 +273,352 @@ def test_sensitivity_on_aggregated_any_index(simple_trace):
+        remaining_calls=0,
+        time_to_next_ms=None,
+        output_tokens=10,
+        span_start_time=start,
+        span_end_time=end,
+    )
+
+
+def test_logical_positions_transitive_overlap():
+    """Transitive overlaps must be collapsed into one group.
+
+    A(0–10) overlaps B(1–3) and C(4–6).  B and C do not overlap each other
```

Ruff (0.15.2)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)